researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -580,22 +577,20 @@ private SchedulesAndTasksModel translateActivities(@NonNull ScheduledActivityLis
         logger.info("translateActivities()");
 
         // first, group activities by day
-        Map<Integer, List<ScheduledActivity>> activityMap = new HashMap<>();
+        ListMultimap<Integer, ScheduledActivity> activityMap = LinkedListMultimap.create();
         for (ScheduledActivity sa : activityList.getItems()) {
             int day = sa.getScheduledOn().dayOfYear().get();
-            List<ScheduledActivity> actList = activityMap.get(day);
-            if (actList == null) {
-                actList = new ArrayList<>();
-                actList.add(sa);
-                activityMap.put(day, actList);
-            } else {
-                actList.add(sa);
-            }
+            activityMap.put(day, sa);
         }
 
+        // Sort the days. There aren't that many days, so this should be relatively quick.
+        List<Integer> dayList = new ArrayList<>(activityMap.keySet());
+        Collections.sort(dayList);
+
+        // Convert from Bridge activities to ResearchKit task model
         SchedulesAndTasksModel model = new SchedulesAndTasksModel();
         model.schedules = new ArrayList<>();
-        for (int day : activityMap.keySet()) {
+        for (int day : dayList) {
             List<ScheduledActivity> aList = activityMap.get(day);
             ScheduledActivity temp = aList.get(0);
 
```

Comment:
This isn't related to your PR, but I just found that the code in this method has a critical error regarding daylight saving time:
"sm.scheduledOn = temp.getScheduledOn().toDate();"
Will translate the scheduledOn date incorrectly if today is before daylight, and the activity is scheduled after daylight. It should be...
"sm.scheduledOn = temp.getScheduledOn().toLocalDate().toDate();"

Reply:
What's the semantics of ScheduleModel.scheduledOn? This is defined as a java.util.Date, which appears to be a timestamp, not a calendar date. Is it correct to flatten this to midnight?